src/ol_infrastructure/applications/concourse/__main__.py

```diff
@@ -0,0 +1,532 @@
+from ol_infrastructure.lib.aws.iam_helper import IAM_POLICY_VERSION, lint_iam_policy
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.stack_defaults import defaults
+
+concourse_config = Config("concourse")
+stack_info = parse_stack()
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+policy_stack = StackReference("infrastructure.aws.policies")
+dns_stack = StackReference("infrastructure.aws.dns")
+consul_stack = StackReference(f"infrastructure.consul.operations.{stack_info.name}")
+mitodl_zone_id = dns_stack.require_output("odl_zone_id")
+
+consul_security_groups = consul_stack.require_output("security_groups")
+operations_vpc = network_stack.require_output("operations_vpc")
+aws_config = AWSBase(
+    tags={"OU": "operations", "Environment": f"operations-{stack_info.env_suffix}"}
+)
+aws_account = get_caller_identity()
+ops_vpc_id = operations_vpc["id"]
+concourse_web_ami = ec2.get_ami(
+    filters=[
+        ec2.GetAmiFilterArgs(name="name", values=["concourse-web-*"]),
+        ec2.GetAmiFilterArgs(name="virtualization-type", values=["hvm"]),
+        ec2.GetAmiFilterArgs(name="root-device-type", values=["ebs"]),
+    ],
+    most_recent=True,
+    owners=[aws_account.account_id],
+)
+
+concourse_worker_ami = ec2.get_ami(
+    filters=[
+        ec2.GetAmiFilterArgs(name="name", values=["concourse-worker-*"]),
+        ec2.GetAmiFilterArgs(name="virtualization-type", values=["hvm"]),
+        ec2.GetAmiFilterArgs(name="root-device-type", values=["ebs"]),
+    ],
+    most_recent=True,
+    owners=[aws_account.account_id],
+)
+
+###################################
+#    Security & Access Control    #
+###################################
+
+# AWS Permissions Document
+# S3 bucket permissions for publishing OCW
+# S3 bucket permissions for uploading software artifacts
+concourse_iam_permissions = {
+    "Version": IAM_POLICY_VERSION,
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "s3:ListAllMyBuckets",
+            "Resource": "*",
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetObject",
+                "s3:PutObject",
+                "s3:DeleteObject",
+                "s3:ListBucket*",
+            ],
+            "Resource": [
+                "arn:aws:s3:::ocw-content*",
+                "arn:aws:s3:::ocw-content*/*",
+                "arn:aws:s3:::ol-eng-artifacts" "arn:aws:s3:::ol-eng-artifacts/*",
+            ],
+        },
+    ],
+}
+
+concourse_iam_policy = iam.Policy(
+    "cicd-iam-permissions-policy",
+    path=f"/ol-infrastructure/iam/cicd-{stack_info.env_suffix}/",
+    policy=lint_iam_policy(concourse_iam_permissions, stringify=True),
+    name_prefix=f"cicd-policy-{stack_info.env_suffix}",
+    tags=aws_config.tags,
+)
+
+# IAM and instance profile
+concourse_instance_role = iam.Role(
```

Comment:
Web and worker have the same instance role?

Reply:
Yes, for now